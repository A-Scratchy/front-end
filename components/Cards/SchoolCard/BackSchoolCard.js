import React from 'react';
import PropTypes from 'prop-types';
import CloseButton from 'components/_common_/CloseButton/CloseButton';
import styles from './BackSchoolCard.css';

BackSchoolCard.propTypes = {
  cardFlipCallback: PropTypes.func.isRequired,
  locations: PropTypes.arrayOf(
    PropTypes.shape({
      city: PropTypes.string,
      va_accepted: PropTypes.bool.isRequired,
      state: PropTypes.string,
    }),
  ).isRequired,
  logoSource: PropTypes.string.isRequired,
  schoolName: PropTypes.string.isRequired,
};

function BackSchoolCard({ cardFlipCallback, locations, logoSource, schoolName }) {
  return (
    <>
      <CloseButton onClick={cardFlipCallback} theme="white" />
      <img src={logoSource} alt={`${schoolName} logo`} />
      <span className={styles.disclaimer}>
        <b>*</b> denotes a location that accepts the GI Bill
      </span>
      <hr className={styles.divider} />
      <ul className={styles.locations}>
        {locations.map(location => (
          <li key={`${location}`}>
            {location.city && location.state && `${location.city}, ${location.state}`}
            {location.va_accepted ? '*' : null}
          </li>
        ))}
      </ul>
    </>
  );
}

export default BackSchoolCard;
