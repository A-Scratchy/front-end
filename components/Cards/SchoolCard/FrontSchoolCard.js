import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Button from 'components/_common_/Button/Button';
import OutboundLink from 'components/_common_/OutboundLink/OutboundLink';
import DesktopIcon from 'static/images/icons/FontAwesome/desktop-solid.svg';
import DevicesIcon from 'static/images/icons/FontAwesome/devices-alt-solid.svg';
import HomeIcon from 'static/images/icons/FontAwesome/home-solid.svg';
import PeopleIcon from 'static/images/icons/FontAwesome/users-solid.svg';
import GIBillApprovedIcon from 'static/images/icons/gi-bill-approved.svg';
import GIBillUnavailableIcon from 'static/images/icons/gi-bill-unavailable.svg';
import Badge from 'components/Badge/Badge';
import styles from './FrontSchoolCard.css';

export default class FrontSchoolCard extends Component {
  static propTypes = {
    cardFlipCallback: PropTypes.func.isRequired,
    hasHardwareIncluded: PropTypes.bool.isRequired,
    hasHousing: PropTypes.bool.isRequired,
    hasOnline: PropTypes.bool.isRequired,
    hasOnlyOnline: PropTypes.bool.isRequired,
    isFullTime: PropTypes.bool.isRequired,
    locations: PropTypes.arrayOf(
      PropTypes.shape({
        city: PropTypes.string,
        va_accepted: PropTypes.bool.isRequired,
        state: PropTypes.string,
      }),
    ).isRequired,
    logoSource: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    website: PropTypes.string.isRequired,
  };

  render() {
    const { props } = this;

    return (
      <>
        <h5>{props.name}</h5>
        <img src={props.logoSource} alt="" />

        <div className={styles.interactionsContainer}>
          <Button>
            <OutboundLink analyticsEventLabel={`${props.name} | Website`} href={props.website}>
              Go To Website
            </OutboundLink>
          </Button>

          {props.locations.length > 1 ? (
            <Button
              analyticsObject={{
                action: 'Button Selected',
                category: 'Interactions',
                label: `${props.name} | Locations`,
              }}
              onClick={props.cardFlipCallback}
            >
              See Locations
            </Button>
          ) : (
            <span className={styles.singleLocationText}>
              {props.hasOnlyOnline
                ? 'Online Only'
                : `${props.locations[0].city}, ${props.locations[0].state}`}
            </span>
          )}
        </div>

        <hr className={styles.divider} />

        <div className={styles.detailsRow}>
          {/* GI Bill */}
          {props.locations.some(location => location.va_accepted) ? (
            <div className={`${styles.detailItem} ${styles.active}`}>
              <Badge
                label="G.I Bill"
                className={styles.codeSchoolCardIcon}
                icon={<GIBillApprovedIcon />}
              />
            </div>
          ) : (
            <div className={`${styles.detailItem} ${styles.disabled}`}>
              <Badge
                label="G.I Bill"
                className={styles.codeSchoolCardIcon}
                icon={<GIBillUnavailableIcon />}
              />
            </div>
          )}

          {/* Online Education */}
          <div
            className={classNames(styles.detailItem, {
              [styles.active]: props.hasOnline,
              [styles.disabled]: !props.hasOnline,
            })}
          >
            <Badge label="Online" className={styles.codeSchoolCardIcon} icon={<DesktopIcon />} />
          </div>

          {/* In-Person Education */}
          <div
            className={classNames(styles.detailItem, {
              [styles.active]: props.isFullTime,
              [styles.disabled]: !props.isFullTime,
            })}
          >
            {' '}
            <Badge label="In-Person" className={styles.codeSchoolCardIcon} icon={<PeopleIcon />} />
          </div>
        </div>

        <div className={styles.detailsRow}>
          {/* Equipment Provided */}
          <div
            className={classNames(styles.detailItem, {
              [styles.active]: props.hasHardwareIncluded,
              [styles.disabled]: !props.hasHardwareIncluded,
            })}
          >
            {' '}
            <Badge label="Equipment" className={styles.codeSchoolCardIcon} icon={<DevicesIcon />} />
          </div>

          {/* Housing Provided */}
          <div
            className={classNames(styles.detailItem, {
              [styles.active]: props.hasHousing,
              [styles.disabled]: !props.hasHousing,
            })}
          >
            {' '}
            <Badge label="Housing" className={styles.codeSchoolCardIcon} icon={<HomeIcon />} />
          </div>
        </div>
      </>
    );
  }
}
