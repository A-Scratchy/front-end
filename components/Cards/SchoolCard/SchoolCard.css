.SchoolCard {
  width: 330px;
  font-family: var(--primaryFontFamily), sans-serif;
  background-color: #ffffff;
  border: 1px solid var(--primary);
  justify-content: normal;
  padding: 0;
  position: relative;
  color: rgba(0, 0, 0, 0.7);
  overflow: hidden;
}

@media screen and (--large-viewport) {
  .SchoolCard {
    width: 300px;
  }
}

@media screen and (--small-viewport) {
  .SchoolCard {
    width: 100%;
  }
}

.giBillRibbon {
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0;
  padding: 0.3rem 0 0.1rem; /* Add more padding on top to fix vertically center */
  width: 55px;
  background: var(--secondary);
  color: var(--primary);
  position: absolute;
  top: 0;
  right: 0;
  text-transform: uppercase;
  text-align: center;
  transform: translateX(20%) translateY(0%) rotate(40deg);
  transform-origin: top left;
}

.giBillRibbon::before,
.giBillRibbon::after {
  content: '';
  position: absolute;
  top: 0;
  margin: 0 -1px;
  width: 100%;
  height: 100%;
  background: var(--secondary);
}

.giBillRibbon::before {
  right: 100%;
}

.giBillRibbon::after {
  left: 100%;
}

.cardBrand {
  text-align: center;
  padding: 0.25rem 0;
}

.SchoolCard,
.cardBlock {
  box-sizing: border-box;
}

.cardBlock {
  display: flex;
  width: 100%;
  padding: 1rem 0.75rem;
  border-top: 1px solid var(--gray);
}

.cardBlockTitle {
  font-weight: bold;
  display: block;
  margin-right: auto;
  padding-right: 6px;
  color: var(--secondary);
}

.modalToggler {
  min-width: auto;
  text-transform: none;
  padding: 0;
  border: 0;
  line-height: inherit;
  text-decoration: underline;
  border-radius: 0;
  letter-spacing: 0;
  background-color: #ffffff;
  margin-left: 5px;
  color: inherit;
}

.badgeGroup {
  display: flex;
}

.badgeGroupItem {
  margin: 0 8px;
  font-size: 15px;
  color: rgba(0, 0, 0, 0.65);
}

.badgeGroupItem:last-child {
  margin-right: 0;
}

.badgeGroupItem svg {
  height: 36px;
  margin: 0;
}

@media screen and (--small-viewport) {
  .badgeGroupItem svg {
    height: 32px;
  }
}

@media screen and (--small-viewport) {
  .badgeGroupItem svg {
    height: 28px;
  }
}

.badgeGroupItem figcaption {
  margin-top: 10px;
}

@media screen and (--small-viewport) {
  .badgeGroupItem figcaption {
    font-size: 0.875em;
  }
}
