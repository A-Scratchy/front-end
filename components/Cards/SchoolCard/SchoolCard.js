import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Card from 'components/Cards/Card/Card';
import OnlineIcon from 'static/images/icons/Custom/online.svg';
import CampusIcon from 'static/images/icons/Custom/campus.svg';
import HousingIcon from 'static/images/icons/Custom/housing.svg';
import EquipmentIcon from 'static/images/icons/Custom/equipment.svg';
import LinkButton from 'components/LinkButton/LinkButton';
import Button from 'components/Button/Button';
import Badge from 'components/Badge/Badge';
import styles from './SchoolCard.css';

class SchoolCard extends Component {
  static propTypes = {
    hasHardwareIncluded: PropTypes.bool.isRequired,
    hasHousing: PropTypes.bool,
    hasOnline: PropTypes.bool.isRequired,
    hasOnlyOnline: PropTypes.bool.isRequired,
    isFullTime: PropTypes.bool.isRequired,
    locations: PropTypes.arrayOf(
      PropTypes.shape({
        city: PropTypes.string,
        va_accepted: PropTypes.bool.isRequired,
        state: PropTypes.string,
      }),
    ).isRequired,
    logoSource: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    website: PropTypes.string.isRequired,
    toggleModal: PropTypes.func.isRequired,
  };

  static defaultProps = {
    hasHousing: true,
  };

  toggleModalClick = () => {
    const { name, locations, toggleModal } = this.props;
    toggleModal({ name, locations });
  };

  render() {
    const { props } = this;

    const hasManyLocations = props.locations.length > 1;

    let locationText = 'Multiple locations';
    if (props.hasOnlyOnline) {
      locationText = 'Online only';
    } else if (!hasManyLocations) {
      const [location] = props.locations;
      locationText = `${location.city}, ${location.state}`;
    }

    const analyticsObject = {
      action: 'Button Selected',
      category: 'Interactions',
      label: `${props.name} | Locations`,
    };

    return (
      <Card className={styles.SchoolCard} hasAnimationOnHover={false}>
        <>
          <div className={styles.giBillRibbon}>GI Bill</div>

          <div className={styles.cardBrand}>
            <img src={props.logoSource} alt={`${props.name} logo`} height="150" />
          </div>

          <div className={styles.cardBlock}>
            <span className={styles.cardBlockTitle}>Availability</span>
            <div className={styles.badgeGroup}>
              {props.hasOnline && (
                <Badge label="Online" icon={<OnlineIcon />} className={styles.badgeGroupItem} />
              )}
              {props.hasOnlyOnline || (
                <Badge label="Campus" icon={<CampusIcon />} className={styles.badgeGroupItem} />
              )}
              {props.hasHousing && (
                <Badge label="Housing" icon={<HousingIcon />} className={styles.badgeGroupItem} />
              )}
              {props.hasHardwareIncluded && (
                <Badge
                  label="Equipment"
                  icon={<EquipmentIcon />}
                  className={styles.badgeGroupItem}
                />
              )}
            </div>
          </div>

          <div className={styles.cardBlock}>
            <span className={styles.cardBlockTitle}>Accepts GI Bill</span>
            Yes
          </div>

          <div className={styles.cardBlock}>
            <span className={styles.cardBlockTitle}>Campus Locations</span>
            {locationText}
            {hasManyLocations && (
              <>
                {' ('}
                <Button
                  analyticsObject={analyticsObject}
                  onClick={this.toggleModalClick}
                  className={styles.modalToggler}
                >
                  view all
                </Button>
                {')'}
              </>
            )}
          </div>

          <div className={styles.cardBlock}>
            <LinkButton
              analyticsEventLabel={`${props.name} | Website`}
              href={props.website}
              fullWidth
            >
              Visit Website
            </LinkButton>
          </div>
        </>
      </Card>
    );
  }
}

export default SchoolCard;
