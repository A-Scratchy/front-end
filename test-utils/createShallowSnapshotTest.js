/* eslint-env jest */
import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers or larger components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
