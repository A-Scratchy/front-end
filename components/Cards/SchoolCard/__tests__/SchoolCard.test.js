import React from 'react';
import { shallow } from 'enzyme';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';
import SchoolCard from '../SchoolCard';

const locations = [
  {
    va_accepted: true,
    address1: '825 Battery Street',
    address2: '3rd Floor',
    city: 'San Francisco',
    zip: '94111',
  },
];

describe('SchoolCard', () => {
  it('should render with required props', () => {
    createShallowSnapshotTest(
      <SchoolCard
        hasHardwareIncluded
        hasHousing
        hasOnline
        hasOnlyOnline={false}
        isFullTime
        locations={locations}
        logoSource="logoSource.com"
        name="school name"
        website="website.com"
      >
        <p>Testing!</p>
      </SchoolCard>,
    );
  });
  it('call to showBackOfCard/showFrontOfTheCard flips isFrontOfCardShowing', () => {
    const wrapper = shallow(
      <SchoolCard
        cardFlipCallback={() => {}}
        hasHardwareIncluded
        hasHousing
        hasOnline
        hasOnlyOnline={false}
        isFullTime
        locations={locations}
        logoSource="source"
        name="school name"
        website="website"
      >
        Test
      </SchoolCard>,
    );
    const instance = wrapper.instance();
    wrapper.setState({ isFrontOfCardShowing: true });
    expect(wrapper.state('isFrontOfCardShowing')).toBe(true);
    instance.showBackOfCard();
    expect(wrapper.state('isFrontOfCardShowing')).toBe(false);
    instance.showFrontOfCard();
    expect(wrapper.state('isFrontOfCardShowing')).toBe(true);
  });
});
