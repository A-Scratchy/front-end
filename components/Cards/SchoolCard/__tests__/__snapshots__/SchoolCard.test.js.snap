// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchoolCard should render with required props 1`] = `
<Card
  className="SchoolCard"
  hasAnimationOnHover={false}
>
  <React.Fragment>
    <div
      className="giBillRibbon"
    >
      GI Bill
    </div>
    <div
      className="cardBrand"
    >
      <img
        alt="school name logo"
        height="150"
        src="source"
      />
    </div>
    <div
      className="cardBlock"
    >
      <span
        className="cardBlockTitle"
      >
        Availability
      </span>
      <div
        className="badgeGroup"
      >
        <Badge
          className="badgeGroupItem"
          icon={<Svg />}
          isImageFirst={true}
          label="Online"
        />
        <Badge
          className="badgeGroupItem"
          icon={<Svg />}
          isImageFirst={true}
          label="Campus"
        />
        <Badge
          className="badgeGroupItem"
          icon={<Svg />}
          isImageFirst={true}
          label="Housing"
        />
        <Badge
          className="badgeGroupItem"
          icon={<Svg />}
          isImageFirst={true}
          label="Equipment"
        />
      </div>
    </div>
    <div
      className="cardBlock"
    >
      <span
        className="cardBlockTitle"
      >
        Accepts GI Bill
      </span>
      Yes
    </div>
    <div
      className="cardBlock"
    >
      <span
        className="cardBlockTitle"
      >
        Campus Locations
      </span>
      Multiple locations
      <React.Fragment>
         (
        <Button
          analyticsObject={
            Object {
              "action": "Button Selected",
              "category": "Interactions",
              "label": "school name | Locations",
            }
          }
          className="modalToggler"
          datum=""
          disabled={false}
          fullWidth={false}
          onClick={[Function]}
          tabIndex={0}
          theme="primary"
          type="button"
        >
          view all
        </Button>
        )
      </React.Fragment>
    </div>
    <div
      className="cardBlock"
    >
      <LinkButton
        analyticsEventLabel="school name | Website"
        disabled={false}
        fullWidth={true}
        href="website"
        theme="primary"
      >
        Visit Website
      </LinkButton>
    </div>
  </React.Fragment>
</Card>
`;
